lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/navigator/Navigator.java

```diff
@@ -266,10 +266,20 @@ CoordinatorLayout getOverlaysLayout() {
     }
 
     public void onHostPause() {
-        super.onViewDisappear();
+        overlayManager.onHostPause();
+        if (!modalStack.isEmpty()) {
+            modalStack.onHostPause();
+        } else {
+            onViewDisappear();
+        }
     }
 
     public void onHostResume() {
-        super.onViewDidAppear();
+        overlayManager.onHostResume();
+        if (!modalStack.isEmpty()) {
```

Comment:
Same as the comment above. If the modal is displayed with  `overCurrentContext`, the root layout also needs to receive an appear event.